components/switch/src/switch.js

```diff
@@ -0,0 +1,165 @@
+/** @jsx jsx */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme, paint } from '@westpac/core';
+
+import { SwitchText, SwitchToggle } from './styled';
+
+// ==============================
+// Utils
+// ==============================
+
+// ==============================
+// Component
+// ==============================
+
+export const Switch = ({
+	name,
+	id,
+	toggleText,
+	size,
+	block,
+	flip,
+	srOnlyText,
+	defaultChecked,
+	disabled,
+	children,
+	...props
+}) => {
+	const { switch: formSwitch } = useTheme();
+
+	const [isChecked, setIsChecked] = useState(defaultChecked);
+	const toggle = () => setIsChecked(!isChecked);
```

Comment:
This should ideally be the onToggle function passed in but in case they dont pass on I suggest doing
```
const toggle = () => {
		if (onToggle) {
			onToggle();
		} else {
			setIsChecked(!isChecked);
		}
	};
```
Need to discuss further with thinkmill about input component control patterns